### PR TITLE
Add densvis and snifter sections

### PR DESCRIPTION
**Pull request by Alanocallaghan**

### CODE REVIEW

**Comment by LTLA** [comment]
I just fiddled with the wording and the format, you can have a look if it still makes sense.

Also, can you add some "unit tests" for the final statement, i.e., 

> the astrocytes, endothelial and microglia clusters being less compact in the density-preserving versions.

Check out the various `stopifnot` calls scattered about the place. Also a good idea to set the seed in each block so as to reduce the fragility of the results to changes in the stream due to randomization elsewhere.

File: inst/book/more-reddim.Rmd [hunk 1]
```diff
@@ -479,8 +480,22 @@ dm <- densmap(reducedDim(sce.zeisel, "PCA"), dens_frac = 0.4, dens_lambda = 0.2)
...
 ```
 
+
+```{r, echo=FALSE}
+ds <- reducedDim(sce.zeisel, "dens-SNE")
+ts <- reducedDim(sce.zeisel, "TSNE")
+
+vars_d <- colVars(ds[sce.zeisel$level1class == "astrocytes_ependymal", ])
+vars_t <- colVars(ts[sce.zeisel$level1class == "astrocytes_ependymal", ])
+stopifnot(mean(vars_d) > mean(vars_t))
```
Comment (LTLA) [inline inst/book/more-reddim.Rmd:490]: Are the two sets of coordinates comparable on the same scale? Should we normalize the variances by, e.g., the range of the plot?

**Comment by Alanocallaghan** [comment]
Good shouts. Still makes sense to me, though I'm a bit upset that you stomped all over my 80 character linewidth. There was an undefined figure ref as far as I can tell tho

There's some odd outlier behaviour that means qualitatively the compactness thing is probably true for other clusters but numerically not really, and I'm not about to start doing quantiles to fudge it, so I've dialed back to  just commenting about astrocytes

**Comment by LTLA** [comment]
> Still makes sense to me, though I'm a bit upset that you stomped all over my 80 character linewidth

Oh yeah, I forget that other people don't have curved monitors.

Anyway, I'm just going to assume it runs fine because BioC's **snifter** is still doing its odd thing with the empty tarballs. The build system will throw it against the wall on Friday, let's see if it sticks.

**Comment by Alanocallaghan** [comment]
Actually I do as well, I just tend to keep 7 windows and a few tabs squeezed into the available horizontal space

Cheers, I too like to live dangerously

**Pull request merged**